pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -637,6 +637,20 @@ func (k *Kaniko) resolveFailFast(ctx context.Context,
 				continue
 			}
 			if jobPod.Status.Phase == v1.PodPending || jobPod.Status.Phase == v1.PodUnknown {
+				if failure, failed := k.getLastPodWarningEvent(ctx, namespace, jobPod.Name); failed {
+					errorMessage := fmt.Sprintf("%s event for Kaniko pod %s. Message: %s",
+						failure.Reason,
+						jobPod.Name,
+						failure.Message)
+
+					// if an error has changed, print it to the logs
+					if errorMessage != lastError {
+						buildLogger.WarnWithCtx(ctx, fmt.Sprintf("%s event for Kaniko pod", failure.Reason),
+							"podName", jobPod.Name,
+							"reason", failure.Message)
```

Comment:
this is a bit confusing, because `reason` is not the failure `Reason`. how about `eventMessage`?